From 39e7e177473350b3a5c34e8824af3b98e25efa89 Mon Sep 17 00:00:00 2001
From: Max Wang <mwang@fb.com>
Date: Fri, 1 Jul 2016 09:08:51 -0700
Subject: [PATCH] Fix param types for mcrypt_get_block_size() to match PHP

Reviewed By: edwinsmith

Differential Revision: D3447281

fbshipit-source-id: a88549307faad886f5a0f78d421aa28e08d21b1c
---
 hphp/runtime/ext/mcrypt/ext_mcrypt.cpp           | 4 ++--
 hphp/runtime/ext/mcrypt/ext_mcrypt.php           | 2 +-
 hphp/test/slow/ext_mcrypt/ext_mcrypt.php         | 1 +
 hphp/test/slow/ext_mcrypt/ext_mcrypt.php.expectf | 2 ++
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/hphp/runtime/ext/mcrypt/ext_mcrypt.cpp b/hphp/runtime/ext/mcrypt/ext_mcrypt.cpp
index 0d6f308acf8e..4e0413dda602 100644
--- a/hphp/runtime/ext/mcrypt/ext_mcrypt.cpp
+++ b/hphp/runtime/ext/mcrypt/ext_mcrypt.cpp
@@ -457,10 +457,10 @@ Variant HHVM_FUNCTION(mcrypt_ofb, const String& cipher, const String& key,
 }
 
 Variant HHVM_FUNCTION(mcrypt_get_block_size, const String& cipher,
-                                    const Variant& module /* = null_string */) {
+                                             const String& mode) {
   MCRYPT td = mcrypt_module_open((char*)cipher.data(),
                                  (char*)MCG(algorithms_dir).data(),
-                                 (char*)module.asCStrRef().data(),
+                                 (char*)mode.data(),
                                  (char*)MCG(modes_dir).data());
   if (td == MCRYPT_FAILED) {
     MCRYPT_OPEN_MODULE_FAILED("mcrypt_get_block_size");
diff --git a/hphp/runtime/ext/mcrypt/ext_mcrypt.php b/hphp/runtime/ext/mcrypt/ext_mcrypt.php
index 39c0583fdd4a..c447deb77c75 100644
--- a/hphp/runtime/ext/mcrypt/ext_mcrypt.php
+++ b/hphp/runtime/ext/mcrypt/ext_mcrypt.php
@@ -297,7 +297,7 @@ function mcrypt_generic(resource $td,
  */
 <<__Native>>
 function mcrypt_get_block_size(string $cipher,
-                               ?string $mode = null): mixed;
+                               string $mode): mixed;
 
 /**
  * Gets the name of the specified cipher
diff --git a/hphp/test/slow/ext_mcrypt/ext_mcrypt.php b/hphp/test/slow/ext_mcrypt/ext_mcrypt.php
index d55dc9f0e631..d44ebd148c73 100644
--- a/hphp/test/slow/ext_mcrypt/ext_mcrypt.php
+++ b/hphp/test/slow/ext_mcrypt/ext_mcrypt.php
@@ -106,6 +106,7 @@ function VERIFY($x) { VS($x != false, true); }
 //////////////////////////////////////////////////////////////////////
 
 VS(mcrypt_get_block_size("tripledes", "ecb"), 8);
+mcrypt_get_block_size("tripledes");
 VS(mcrypt_get_cipher_name(MCRYPT_TRIPLEDES), "3DES");
 VS(mcrypt_get_iv_size(MCRYPT_CAST_256, MCRYPT_MODE_CFB), 16);
 VS(mcrypt_get_iv_size("des", "ecb"), 8);
diff --git a/hphp/test/slow/ext_mcrypt/ext_mcrypt.php.expectf b/hphp/test/slow/ext_mcrypt/ext_mcrypt.php.expectf
index bdfd53d2ba4f..4d3ac560b89c 100644
--- a/hphp/test/slow/ext_mcrypt/ext_mcrypt.php.expectf
+++ b/hphp/test/slow/ext_mcrypt/ext_mcrypt.php.expectf
@@ -38,6 +38,8 @@ Deprecated: Function mcrypt_ofb() is deprecated in %s/test/slow/ext_mcrypt/ext_m
 bool(true)
 bool(true)
 bool(true)
+
+Warning: mcrypt_get_block_size() expects exactly 2 parameters, 1 given in %s/test/slow/ext_mcrypt/ext_mcrypt.php on line 109
 bool(true)
 bool(true)
 bool(true)
